src/poll.rs

```diff
@@ -197,8 +187,8 @@ impl PollWatcher {
 }
 
 impl Watcher for PollWatcher {
-    fn new_immediate(tx: Sender<Result<Event>>) -> Result<PollWatcher> {
-        PollWatcher::with_delay(tx, Duration::from_secs(30))
+    fn new_immediate<F: EventFn>(event_fn: F) -> Result<PollWatcher> {
+        PollWatcher::with_delay(event_fn, Duration::from_secs(30))
     }
 
     fn watch<P: AsRef<Path>>(&mut self, path: P, recursive_mode: RecursiveMode) -> Result<()> {
```

Comment:
Unrelated, but this suffers from the same problem: it's a large generic function which is monomorphised for every `P`.